### PR TITLE
Fix `sandbox` call on 1.11-beta

### PR DESCRIPTION
**Pull request by kbarros**
As a possible fix to https://github.com/JuliaTesting/TestEnv.jl/issues/86, one could modify the arguments to `sandbox` at the call site, depending on a static version comparison. This would minimize code duplication by reusing the rest of the code inside `src/julia-1.9`. In this branch, tests pass for me locally on both 1.10 and 1.11-beta.

### CODE REVIEW

**Comment by kbarros** [comment]
Is this blocked on the Codecov failure? This PR just removes one of the arguments to `sandbox` when Julia 1.11 is detected. Any suggestions for pushing the PR forward?

**Comment by oxinabox** [comment]
Sorry no this is blocked by me not seeing it til now.
Don't hesitate to bump PRs after a week or so, its very easy for people maintaining many packages to not see a notification/email.

Having given this some thought, 
I think the better thing to do is to copy julia-1.9 into a julia-1.11 folder.
We used to have a lot of these branches in the code before and it got pretty confusing once you get enough of them.
As I wouldn't be surprised to need more for 1.11, with the new workspaces functionality.
(But we won't see that til some one tried TestEnv in somehing using it extensively and it breaks).

**Comment by kbarros** [comment]
I have updated the PR as you suggested. The new commits undo all modifications to the `julia-1.9/` folder, and create a new directory `julia-1.11/` with the minimal fixes. Tests pass locally on 1.10 and 1.11.0-beta2. Merging this PR will hopefully unblock https://github.com/julia-vscode/TestItemRunner.jl/issues/72.

**Comment by davidanthoff** [comment]
@oxinabox ready to merge now?

**Review by davidanthoff — APPROVED** [review]

**Comment by kbarros** [comment]
Gentle ping

**Comment by oxinabox** [comment]
Sorry I was away for a bit.
Will add David as a comaintainer. I thought I did so a year ago.

This needs to also add 1.10 and 1.11 to the CI matrix

**Comment by kbarros** [comment]
I tried naively adding 1.11 to the CI matrix. This throws an error, since 1.11 hasn't been released. How to proceed?

**Review by davidanthoff — APPROVED** [review]

**Comment by davidanthoff** [comment]
@oxinabox I think I still don't have write access so that I can merge things?

**Comment by oxinabox** [comment]
I am good for this to be merged, I am going to leave it to David to actually merge because I want to check I setup his permissions right, since i failed the first time

**Pull request merged**
